grails-app/services/org/pih/warehouse/order/OrderService.groovy

```diff
@@ -113,12 +113,12 @@ class OrderService {
                     if (params.createdBy) {
                         eq("createdBy.id", params.createdBy)
                     }
-                    if (params.paymentTerm) {
-                        or {
-                            if (params.list("paymentTerm").contains("null")) {
-                                isNull("paymentTerm")
-                            }
-                            paymentTerm(JoinType.LEFT_OUTER_JOIN.joinTypeValue) {
+                    paymentTerm(JoinType.LEFT_OUTER_JOIN.joinTypeValue) {
```

Comment:
why has it been pulled out of the `if`?
I would say it is not necessary to join this table if we don't specify `params.paymentTerm`? Or am I missing something here?

Reply:
if you scroll down few llines and look at the **order(...)**  logic, you will see that we also support ordering by **productTerm**
```groovy
paymentTerm {
    order("name", params.order ?: 'asc')
}
```
if you dislike my approach of pulling out the association with a join then I will have to specify the joint type on the paymentTerm for ordering too like:
```groovy
paymentTerm(JoinType.LEFT_OUTER_JOIN.joinTypeValue) {
    order("name", params.order ?: 'asc')
}
```
what do you think? Should we have the join type defined in both places?

Reply:
This is kind of explanation I expected to get :laughing: 
I have not seen that closure at the bottom. If that's the case, then I'm completely fine with the change.